src/core_modules/capture-core/components/Pages/New/RegistrationDataEntry/RegistrationDataEntry.component.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { useScopeInfo } from '../../../../hooks/useScopeInfo';
+import { scopeTypes } from '../../../../metaData';
+import { startDataEntryInitialisation } from './RegistrationDataEntry.actions';
+import { EnrollmentDataEntry, TrackedEntityInstanceDataEntry } from '../../../DataEntries';
+import { useCurrentOrgUnitInfo } from '../../../../hooks/useCurrentOrgUnitInfo';
+import type { OwnProps } from './RegistrationDataEntry.types';
+import { useRegistrationFormInfoForSelectedScope } from './useRegistrationFormInfoForSelectedScope';
+
+const useDataEntryLifecycle = (selectedScopeId, dataEntryId) => {
+    const dispatch = useDispatch();
+    const selectedOrgUnitInfo = useCurrentOrgUnitInfo();
+    const { formId, formFoundation } = useRegistrationFormInfoForSelectedScope(selectedScopeId);
+
+    useEffect(() => {
+        if (formId) {
+            dispatch(startDataEntryInitialisation({ selectedOrgUnitInfo, dataEntryId, formFoundation }));
```

Comment:
I will change this in a future PR. I dont think this is the best way to initialise this component. But I would love to get your opinion on this